expfmt/openmetrics_parse_test.go

```diff
@@ -0,0 +1,307 @@
+# EOF
+`,
+
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("foo"),
+					Type: dto.MetricType_COUNTER.Enum(),
+				},
+			},
+		},
+
+		// 2: has type and unit as input
+		{
+			in: `# TYPE foo_seconds counter
+# UNIT foo_seconds seconds
+# EOF
+`,
+
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("foo_seconds"),
+					Type: dto.MetricType_COUNTER.Enum(),
+					Unit: proto.String("seconds"),
+				},
+			},
+		},
+
+		// 3: has type, unit, help as input
+		{
+			in: `# HELP foo_seconds abc
+# TYPE foo_seconds counter
+# UNIT foo_seconds seconds
+# EOF
+`,
+
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("foo_seconds"),
+					Type: dto.MetricType_COUNTER.Enum(),
+					Unit: proto.String("seconds"),
+					Help: proto.String("abc"),
+				},
+			},
+		},
+
+		// 4: type gauge
+		{
+			in: `# HELP foo_seconds abc
+# TYPE foo_seconds gauge
+# UNIT foo_seconds seconds
+# EOF
+`,
+
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("foo_seconds"),
+					Type: dto.MetricType_GAUGE.Enum(),
+					Unit: proto.String("seconds"),
+					Help: proto.String("abc"),
+				},
+			},
+		},
+
+		// 5: type histogram
+		{
+			in: `# HELP foo_seconds abc
+# TYPE foo_seconds histogram
+# UNIT foo_seconds seconds
+# EOF
+`,
+
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("foo_seconds"),
+					Type: dto.MetricType_HISTOGRAM.Enum(),
+					Unit: proto.String("seconds"),
+					Help: proto.String("abc"),
+				},
+			},
+		},
+
+		// 6: type summary
+		{
+			in: `# HELP foo_seconds abc
+# TYPE foo_seconds summary
+# UNIT foo_seconds seconds
+# EOF
+`,
+
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("foo_seconds"),
+					Type: dto.MetricType_SUMMARY.Enum(),
+					Unit: proto.String("seconds"),
+					Help: proto.String("abc"),
+				},
+			},
+		},
+
+		// 7: type summary
+		{
+			in: `# HELP foo_seconds abc
+# TYPE foo_seconds gaugehistogram
+# UNIT foo_seconds seconds
+# EOF
+`,
+
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("foo_seconds"),
+					Type: dto.MetricType_GAUGE_HISTOGRAM.Enum(),
+					Unit: proto.String("seconds"),
+					Help: proto.String("abc"),
+				},
+			},
+		},
+
+		// 8: a normal comment
+		{
+			in: `#
+# TYPE name_seconds counter
+# UNIT name_seconds seconds
+# HELP name_seconds two-line\n doc  str\\ing
+# HELP  name2 doc str"ing 2
+#    TYPE    name2 gauge
+# EOF
+`,
+
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("name_seconds"),
+					Type: dto.MetricType_COUNTER.Enum(),
+					Unit: proto.String("seconds"),
+					Help: proto.String("two-line\n doc  str\\ing"),
+				},
+				{
+					Name: proto.String("name2"),
+					Type: dto.MetricType_GAUGE.Enum(),
+					Help: proto.String("doc str\"ing 2"),
+				},
+			},
+		},
+	}
+
+	for i, scenario := range scenarios {
+		out, err := omParser.OpenMetricsToMetricFamilies(strings.NewReader(scenario.in))
+		if err != nil {
+			t.Errorf("%d. error: %s", i, err)
+			continue
+		}
```

Comment:
What do you think about using testify `assert` or `require`? I think it should be more readable, right? We could also add `name` to the scenario struct and use `t.Run(scenario.name...` instead of comments with numbers to make test results more readable.

Reply:
We already import testify in our go.mod, so this should be fine